src/UI/components/sources/source_form.tsx

```diff
@@ -125,6 +126,27 @@ export default function SourceForm() {
           </div>
           <br />
 
+          <div>
+            <Controller
+              name="citation"
+              control={control}
+              render={({
+                field: { onChange, value },
+                fieldState: { error },
+              }) => (
+                <TextField
+                  value={value || ''}
+                  label={'Citation:'}
+                  error={!!error}
+                  helperText={error ? error.message : null}
+                  {...register('citation')}
+                ></TextField>
+              )}
+              rules={{ required: 'Citation required' }}
+            />
+          </div>
```

Comment:
Happy to keep this in for now, but citations should really be generated by the system. There's a story somewhere to sort all the citation stuff out.